src/App.tsx

```diff
@@ -1,61 +1,154 @@
+import { useState } from 'react';
+import classNames from 'classnames';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todos } from './types/Todos';
+import { User } from './types/User';
+
+const todosWithUser = todosFromServer.map(todo => {
+  const userForTodo = usersFromServer.find(user => todo.userId === user.id);
+
+  return {
+    ...todo,
+    user: userForTodo as User,
+  };
+});
 
 export const App = () => {
+  const [allTodos, setAllTodos] = useState(todosWithUser);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const findUser = (id: number) => {
+    return usersFromServer.find(user => user.id === id) as User;
+  };
+
+  const addNewTodo = (newTodo: Todos) => {
+    setAllTodos(todos => [...todos, newTodo]);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const resetFields = () => {
+    setTitle('');
+    setUserId(0);
+    setHasTitleError(false);
+    setHasUserIdError(false);
+  };
+
+  const getNewId = (todos: Todos[]) => {
+    const maxId = Math.max(
+      ...todos.map(todo => todo.id),
+    );
+
+    return maxId + 1;
+  };
```

Comment:
It is more correct to move the function outside the component so that it is not recalculated on each re-render.